src/app/private/scripts/controllers/announcement/createAnnouncementController.js

```diff
@@ -125,12 +114,12 @@ angular.module('playerApp').controller('createAnnouncementCtrl', ['$rootScope',
      * @desc - add new url input box
      * @memberOf Controllers.createAnnouncementCtrl
      */
-    createAnn.addNewLink = function () {
-      var newItemNo = createAnn.repeatableWebLinks.length + 1
-      createAnn.repeatableWebLinks.push({
+    composeAnn.addNewLink = function () {
+      var newItemNo = composeAnn.repeatableWebLinks.length + 1
+      composeAnn.repeatableWebLinks.push({
```

Comment:
this state change should be made on the AnnouncementModel not on the controller state. please fix.